_source/_posts/2022-08-19-build-crud-spring-and-vue.md

```diff
@@ -0,0 +1,1057 @@
+
+```bash
+curl https://start.spring.io/starter.tgz \
+  -d bootVersion=2.6.10 \
+  -d javaVersion=11 \
+  -d dependencies=web,data-rest,lombok,data-jpa,h2,okta \
+  -d type=gradle-project \
+  -d baseDir=resource-server \
+| tar -xzvf - && cd resource-server
+```
+
+The dependencies you're including are:
+
+- `web`: [Spring Web MVC](https://docs.spring.io/spring-framework/docs/3.2.x/spring-framework-reference/html/mvc.html), adds basic HTTP REST functionality
+- `data-jpa`: [Spring Data JPA](https://spring.io/projects/spring-data-jpa), makes it easy to create JPA-based repositories
+- `data-rest`: [Spring Data REST](https://spring.io/projects/spring-data-rest), exposes Spring Data repositories as resource servers
+- `h2`: the [H2](https://www.h2database.com/html/main.html) in-memory database used for demonstration purposes
+- `lombok`: [Project Lombok](https://projectlombok.org/), adds some helpful annotations that eliminate the need to write a lot of getters and setters
+- `okta`: [Okta Spring Boot Starter](https://github.com/okta/okta-spring-boot) that helps OAuth 2.0 and OIDC configuration
+
+Project Lombok saves a lot of clutter and ceremony code. However, if you're using an IDE, **you'll need to install a plugin for Lombok**.
+
+## Update the Secure Spring Boot app
+
+Open the application properties file and update it. You're changing the server port so it doesn't conflict with the default Vue local server (which also defaults `8080`). 
+
+`src/main/resources/application.properties`
+
+```properties
+server.port=9000
+okta.oauth2.issuer=<your-issuer-uri>
+okta.oauth2.clientId=<your-client-id>
+```
+
+**You need to replace the two bracketed values** with the values you generated above for the OIDC app using the Okta CLI.
+
+You can run the bootstrapped project right now and see if it starts. It should start but won't do much.
+
+```bash
+./gradlew bootRun
+```
+
+Create a `SecurityConfiguration` class to configure Spring Security. The class below configures web security to allow all requests, effectively bypassing security. This is just so you can test the resource server initially. You'll enable security shortly.
+
+`src/main/java/com/example/demo/SecurityConfiguration.java`
+
+```java
+package com.example.demo;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.web.SecurityFilterChain;
+
+@EnableWebSecurity
+@Configuration
+public class SecurityConfiguration {
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http.authorizeRequests()
+                .anyRequest().permitAll();
+        return http.build();
```

Comment:
My IntelliJ formatting is a bit different. 

```java
    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
        http.authorizeRequests()
            .anyRequest().permitAll();
        return http.build();
    }
```

If you're using the defaults, it's fine to leave it as-is, but I think mine looks a little better.